### ISSUE TITLE
Restore-DbaDatabase from Backup-DbaDatabase: Problem with -Verify

### ISSUE THREAD

**Issue by Zhunya** [sqlcollaborative/dbatools#1343]
## Is this:

- [x] *bug*:

## System Details

```
Name                           Value
----                           -----
PSVersion                      5.1.15063.296
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.15063.296
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

- Output of dbatools version:

```
0.8.957
```
Version from #1335

- SQL Server version for source/target
SQL Server 2008 R2

#### Example:
1. Run Command:

```powershell
Backup-DbaDatabase -SqlInstance srv-db -Database DB -CompressBackup -Checksum -Verify -Verbose

```
2. Received following error

```powershell
WARNING: [Backup-DbaDatabase][16:15:37] Backup Failed:  A parameter cannot be found that matches parameter name 'and'.

```

3. Verbose output of same command

```
VERBOSE: [Backup-DbaDatabase][16:14:55] Connecting to srv-db
VERBOSE: [Backup-DbaDatabase][16:14:55] 1 database to backup
VERBOSE: [Backup-DbaDatabase][16:14:55] Backup up database [DD]
VERBOSE: [Backup-DbaDatabase][16:14:55] Compression enabled
VERBOSE: Setting type
VERBOSE: [Backup-DbaDatabase][16:14:55] Sorting Paths
VERBOSE: [Backup-DbaDatabase][16:14:55] Setting filename
VERBOSE: [Backup-DbaDatabase][16:14:55] Devices added
VERBOSE: [Restore-DbaDatabase][16:15:37] Unverified input, full scans
VERBOSE: [Restore-DbaDatabase][16:15:37] Working remotely, and non UNC path used. Dropping to XpDirTree, all paths evaluated at
SRV-DB
VERBOSE: [Restore-DbaDatabase][16:15:37] Only XpDirTree is safe on remote server
VERBOSE: [Restore-DbaDatabase][16:15:37] type = string
VERBOSE:  : Paths passed in
WARNING: [Backup-DbaDatabase][16:15:37] Backup Failed:  A parameter cannot be found that matches parameter name 'and'.

```

```
SqlInstance      : srv-db
DatabaseName     : DB
BackupComplete   : **False**
BackupFilesCount : 1
BackupFile       : DB_201705291638.bak
BackupFolder     : K:\
BackupPath       : K:\DB_201705291638.bak
Script           : BACKUP DATABASE [DB] TO  DISK = N'K:\DB_201705291638.bak' WITH  COPY_ONLY, NOFORMAT, NOINIT, NOSKIP, REWIND, NOUNLOAD, COMPRESSION,  STATS = 1, CHECKSUM
Verified         : **False**
Type             : Full

```

Something went wrong, and at this moment I can't debug, so please @Stuart-Moore look at this problem. Problem some where in Restore-DbaDatabase or in other function called by this one. Without **-Verify** backup finished just fine.

**Comment by Zhunya** [comment]
Oh Have found line with error =)

```
Test-DbaSqlPath : A parameter cannot be found that matches parameter name 'and'.
At line:397 char:118
+ ...  -SqlInstance $SqlInstance -SqlCredential $SqlCredential -and $p -not ...
+                                                              ~~~~
    + CategoryInfo          : InvalidArgument: (:) [Test-DbaSqlPath], ParameterBindingException
    + FullyQualifiedErrorId : NamedParameterNotFound,Test-DbaSqlPath

```

**Comment by Zhunya** [comment]
Fixed in commit #291ee6b
